cmd/spiderpool-agent/cmd/daemon.go

```diff
@@ -157,6 +157,10 @@ func DaemonMain() {
 			logger.Fatal(err.Error())
 		}
 	}()
+	waitForCacheSync := mgr.GetCache().WaitForCacheSync(agentContext.InnerCtx)
+	if !waitForCacheSync {
+		logger.Fatal("failed to wait for syncing controller-runtime cache")
+	}
 
 	logger.Info("Begin to initialize spiderpool-agent OpenAPI HTTP server")
 	srv, err := newAgentOpenAPIHttpServer()
```

Comment:
The added code looks good and ensures that the `waitForCacheSync` method is called before initializing the `spiderpool-agent OpenAPI HTTP server`. However, the context in which this code is executed is important for identifying potential bugs or issues.

Here are some suggestions that might improve the code review:

- It would be helpful to ensure that any panics or unexpected errors from the cache sync or open API initialization are properly caught and managed.
- Consider adding more informative error messages, rather than simply logging a fatal error. This can help in diagnosing any issues that arise during runtime.
- Ensure that the `WaitForCacheSync` method does not introduce any performance overheads, as it might delay the startup time of the server.